ppoprf/examples/client.rs

```diff
@@ -0,0 +1,70 @@
+//! Example clieant for the ppoprf randomness web service.
+//!
+//! This tests and demonstrates the ppoprf evaluation function
+//! in an Actix-Web service application by making example queries.
+//!
+//! To verify the example works, start the server in one terminal:
+//! ```sh
+//! cargo run --example server
+//! ```
+//!
+//! In another terminal, launch the client:
+//! ```sh
+//! cargo run --example client
+//! ```
+
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_COMPRESSED;
+use env_logger::Env;
+use log::info;
+use reqwest::blocking::{Client, get};
+use serde::Serialize;
+
+/// Fetch the server identification string.
+///
+/// Acts as a basic availability ping.
+fn fetch_ident(url: &str) -> reqwest::Result<()> {
+    let res = get(url)?;
+    let status = res.status();
+    let text = res.text()?;
+
+    info!("{} - {}", status, text);
+
+    Ok(())
+}
+
+/// Explicit query body.
+#[derive(Serialize)]
+struct Query {
+    name: String,
+    points: Vec<String>,
+}
+
+/// Fetch randomness from the server.
+///
+/// Acts as a basic round-trip test.
+fn fetch_randomness(url: &str) -> reqwest::Result<()> {
+    let query = Query {
+        name: "example client".into(),
+        points: vec![ base64::encode(RISTRETTO_BASEPOINT_COMPRESSED.0), ],
```

Comment:
I haven't explained this previously, but the clients have to sample their points in the way defined by [`Client::blind`](https://github.com/brave-experiments/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L162-L169).

```
    let mut hashed_input = [0u8; 64];
    strobe_hash(input, "ppoprf_derive_client_input", &mut hashed_input);
    let point = RistrettoPoint::from_uniform_bytes(&hashed_input);
    let mut csprng = OsRng;
    let r = Scalar::random(&mut csprng);
    ((r * point).compress(), r)
```

The LHS of the tuple is the blinded point which is sent to the server, the right hand side is used to unblind the result of the evaluation. The client then receives an `Evaluation` struct from the server. They need to verify the `proof` element using [`Client::verify`](https://github.com/brave-experiments/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L171-L183), and then run [`Client::unblind`](https://github.com/brave-experiments/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L185-L189) where the first input is the point from Evaluation.0, and the second input is the RHS `r` in the tuple output by `Client::blind`.

Let me know if this does not make sense. An example of how this works is given in the [`end_to_end_verify`](https://github.com/brave-experiments/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L307-L317) test function.

Reply:
I added `Client::blind` and `Client::unblind` calls, but finalisation and proof verification require the server publish keys and md values; I haven't implemented that part.

Reply:
> Do I understand correctly that this "works" in the sense that the server can't tell this from any actual blinded point, it's just that it wouldn't work for star submission?

Yes, this is for functionality reasons related to STAR. The client can submit whatever they want, but for STAR it must be a specific value to get the correct evaluation back.

Reply:
LGTM (modulo the changes required to incorporate the usage of metadata and finalization step 👍